src/gwt/src/org/rstudio/studio/client/workbench/views/viewer/ViewerPane.java

```diff
@@ -232,6 +232,15 @@ public void popout()
       {
          globalDisplay_.showHtmlFile(rmdPreviewParams_.getOutputFile());
       }
+      else if (frame_ != null &&
+          !StringUtil.equals(frame_.getIFrame().getCurrentUrl(), getUrl()))
+      {
+         // Typically we navigate to the unmodified URL (i.e. without the
+         // viewer_pane=1 query params, etc.) However, if the URL currently
+         // loaded in the frame is different, the user probably navigated away
+         // from original URL, so load that URL as-is.
+         globalDisplay_.openWindow(frame_.getIFrame().getCurrentUrl());
```

Comment:
Wouldn't this imply an attempt to open a `null` URL if we couldn't get the IFrame URL (e.g. because of DOM restrictions)?

Reply:
Hmm, you're right (I had this test in at some point but lost it when refactoring). Updated!